internal/metrics/metrics.go

```diff
@@ -0,0 +1,114 @@
+package metrics
+
+
+import (
+	"time"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const (
+	ms = 0.001
+	s  = 1000 * ms
+	m  = 60 * s
+	kb = 1024
+	mb = 1024*kb
+)
+
+type metrics struct {
+	registry *prometheus.Registry
+	endpoint string
+
+	// Custom metics
+	request_durations *prometheus.HistogramVec
+	response_sizes    *prometheus.HistogramVec
+
+}
+
+func (m *metrics) GetEndpoint() string {
+	return m.endpoint
+}
+
+/** Create a new metric instance
+ *
+ * The metrics are applies to a new non-global prometheus registry
+ */
+func NewMetrics(endpoint string) *metrics {
+	registry := prometheus.NewRegistry()
+	metrics := &metrics{
+		registry: registry,
+		endpoint: strings.TrimLeft(endpoint, "/"),
+
+		request_durations: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Name:    "vdsslice_durations_histogram_seconds",
+			Help:    "VDSslice latency distributions.",
+			Buckets: []float64{100*ms, 500*ms, 1*s, 2*s, 5*s, 20*s, 1*m, 2*m},
+		}, []string{"method", "path", "status", "cachehit"}),
+
+		response_sizes: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Name:    "vdsslice_response_sizes_histogram_bytes",
+			Help:    "VDSslice response size distributions.",
+			Buckets: []float64{100*kb, 1*mb, 5*mb, 10*mb, 20*mb, 50*mb, 100*mb, 200*mb},
```

Comment:
This thing prints pretty weird stuff like
`status="Internal Server Error",le="1.048576e+06`
This was for wrong inline. Result is 233B.

I think that means that
- we need to add a bucket beneath 1KB.
- 1 unit is a byte, which is good I think, but is there any way that we can format floats without e+06?
- we have to see if there is any way to format the labels, because I would be nice to see "1.1MB" instead of 6 zeroes.